File: core/src/com.tuxbear.dinos/domain/game/LocalGameState.java
package com.tuxbear.dinos.domain.game;

/**
 * Created with IntelliJ IDEA. User: tuxbear Date: 03/01/14 Time: 17:02 To change this template use File | Settings | File
 * Templates.
 */
public enum LocalGameState {
    YOU_CAN_PLAY,
    WAITING_JUST_FOR_YOU,
    WAITING_FOR_OPPONENTS,
    ENDED;


    public String prettyString() {
        switch (this) {
            case YOU_CAN_PLAY:
                return "Your turn";
            case WAITING_JUST_FOR_YOU:
                return "Everyone's waiting!";
            case WAITING_FOR_OPPONENTS:
                return "Waiting";
            case ENDED:
                return "Ended!";
        }

        return "";
    }
}

File: domain/src/com/tuxbear/dinos/domain/game/LocalGameState.kt
package com.tuxbear.dinos.domain.game

enum class LocalGameState {
    YOU_CAN_PLAY, WAITING_JUST_FOR_YOU, WAITING_FOR_OPPONENTS, ENDED;

    fun prettyString(): String {
        return when (this) {
            YOU_CAN_PLAY -> "Game on!"
            WAITING_JUST_FOR_YOU -> "Everyone's waiting!"
            WAITING_FOR_OPPONENTS -> "Waiting"
            ENDED -> "Ended!"
        }
    }
}